lib/local_system.rb

```diff
@@ -105,8 +105,12 @@ def validate_build_compatibility(system_description)
       end
     end
 
+    def matching_architecture?(arch)
```

Comment:
Wouldn't `matches_architecture?` be the better name? Then code using the methods would read naturally as `if system.matches_architecture?`.